File: ReactAndroid/src/main/java/com/facebook/react/bridge/WritableArray.java
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

/**
 * Interface for a mutable array. Used to pass arguments from Java to JS.
 */
public interface WritableArray extends ReadableArray {

  void pushNull();
  void pushBoolean(boolean value);
  void pushDouble(double value);
  void pushInt(int value);
  void pushString(@Nullable String value);
  void pushArray(@Nullable WritableArray array);
  void pushMap(@Nullable WritableMap map);
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/WritableArray.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge

/** Interface for a mutable array. Used to pass arguments from Kotlin to JS. */
public interface WritableArray : ReadableArray {
  public fun pushArray(array: ReadableArray?)

  public fun pushBoolean(value: Boolean)

  public fun pushDouble(value: Double)

  public fun pushInt(value: Int)

  public fun pushLong(value: Long)

  public fun pushMap(map: ReadableMap?)

  public fun pushNull()

  public fun pushString(value: String?)
}
